eclipse/src/saros/editor/internal/EditorAPI.java

```diff
@@ -268,27 +272,275 @@ public static IResource getEditorResource(IEditorPart editorPart) {
   }
 
   /**
-   * Returns the current text selection for given editor.
+   * Returns the line base selection values for the given editor part.
    *
-   * @param editorPart the editorPart for which to get the text selection.
-   * @return the current text selection. Returns {@link TextSelection#emptySelection()} if no text
-   *     selection exists.
+   * <p>The given editor part must not be <code>null</code>.
+   *
+   * @param editorPart the editorPart for which to get the text selection
+   * @return the line base selection values for the given editor part or {@link
+   *     TextSelection#EMPTY_SELECTION} if the given editor part is not a text editor, the editor
+   *     does not have a valid selection provider or document provider, a valid IDocument could not
+   *     be obtained form the document provider, or the correct line numbers or in-lin offsets could
+   *     not be calculated
+   * @throws NullPointerException if the given editor part is <code>null</code>
    */
   public static TextSelection getSelection(IEditorPart editorPart) {
```

Comment:
I know that this is not your fault but the EditorAPI is basically just convenience API regarding Eclipse stuff.

Normally it would be better to return a JFace TextSelection, to be more specific an ITextSelection and then introduce fromCoreToEclipse and fromEclipseToCore methods to convert them.

Reply:
But why? All these methods do is use Eclipse specific API to calculate some values. And yes, the input and output values are wrapped in a class from the "core", but why does it matter? I could also pass them as an array of `int`, but that just seem cumbersome to me.

Under the premise that we are working with the new positioning logic, I don't see a need to access the old selection format. So why keep it in a separate method?

Reply:
As I said it is generic API. Every method beside ```getEditorPath``` is taking and returning Eclipse stuff.

Reply:
So what would you prefer me doing? Create a new utility class named `SelectionAPI` (or something like that) and move the methods there?

Reply:
Just add an TODO. The refactor / movement of the logic is not urgent.

Reply:
Done.